### PR TITLE
886 printing endpoint stub

### PR DESCRIPTION
**Pull request by clemens-msupply**
Adds:
- printReport graphql endpoint that only takes a report id and the id of the entity to be printed, e.g. the invoice id, and returns a file id
- /files?id={id} endpoint to fetch the printed report

The files endpoint already contains some proof of concept logic which needs to be update and put into a service:
- It always creates a dummy txt file for a given id in the static_files directory
- Returns this dummy txt file with a proper file name (not the file id). The "proper" filename is currently encode like `{file_id}_filename.pdf`
- Cleans up old static files that are too old

#886

### CODE REVIEW

File: graphql/printing/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+#[derive(PartialEq, Debug)]
+pub struct PrintReportNode {}
+
+#[Object]
+impl PrintReportNode {
+    /// Return the file id of the printed report.
+    /// The file can be fetched using the /files?id={id} endpoint
+    pub async fn file_id(&self) -> &str {
+        "demofile0123456789"
```
Comment (andreievg) [inline graphql/printing/src/lib.rs:42]: would it be be easier for f/e and easier to change, to have the full url (excluding host) ?

File: server/src/static_files.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        "Static file not found",
+    ))?;
+    let original_file_name = parse_original_file_name(&query.id, &file_path).ok_or(
+        InternalError::new("Invalid file name", StatusCode::INTERNAL_SERVER_ERROR),
+    )?;
+    // clean up the static file directory
+    let max_lifetime_sec = 60 * 60; // 1 hours
+    delete_old_files(file_dir, max_lifetime_sec)?;
+
+    fs::NamedFile::open(file_path)?
```
Comment (andreievg) [inline server/src/static_files.rs:47]: Does this pipe into a response ? 

If not, was wondering if we should just delete the file straight after it's been put into a response

**Review by andreievg — APPROVED** [review]
Looks good thanks

File: server/src/static_files.rs (hunk 2 above)
Comment (clemens-msupply) [reply server/src/static_files.rs:47]: Just had a look, it does stream the content directly from the file so can't delete immediately...

File: graphql/printing/src/lib.rs (hunk 1 above)
Comment (clemens-msupply) [reply graphql/printing/src/lib.rs:42]: My first version was actually:

``` rust
pub async fn file_path(&self) -> &str {
        "/files?id=demofile0123456789&name=test.pdf"
}
```

but I found it a bit odd to use. Returning the whole url including the host would be nicer but difficult to implement. So I decided to only return an id.

**Pull request merged**